### DISCUSSION TITLE
Difference between RT and BT commands

### DISCUSSION THREAD

**Discussion by JoeyNovais** [Accenture/sfmc-devtools#385] [Q&A] [answered]
Hey,

Since version 4.0.0 you provide the `bt` command in addition to the `rt` one.
I think I understood the fact that `rt` takes the source from SFMC while `bt` does it from the retrieve folder.
Is there any "hidden" things behind those commands ?

- Because documentation says for both commands:
> This command is a prerequisite for the buildDefinition command

Do we need to run both commands or just the one we prefer ?
- Also playing with both commands leaded me to this result: 
![image](https://user-images.githubusercontent.com/2325059/179759950-c50966b1-46c5-447a-be1b-9b7af23564fc.png)
So `rt` generate the bottom one and `bt` the one with the prefix token, but I think it only worked like this for assets. Is it intentionnal ?

Thanks

**Comment by JoernBerkefeld** [comment, marked as answer]
Hi there, I believe you identified a bug.
In principle, both should lead to the same result, however, that's clearly not the case here.

- [x] bug: #390 The file should not be renamed by bt

Choosing one or the other command depends on your use case. U understood correctly how they are intended to work. With that said, rt needed to be implemented per type and we never got around to that - means there are types for which it wasn't implemented. 

Bt on the other hand relies on you executing retrieve first and then works for all types you downloaded. It's more generic because it didn't need to handle downloading a single file.

If you want to use templating then you have essentially 3 ways:

1. Use the commands rt, bd(b), d 
2. Use the commands r, bt, bd(b), d
3. Use r, then manually copy & adapt files from retrieve/ to template/, then run bd(b), d

Option 4 is also available and relies on you using a git repo and the command createDeltaPkg & deploy together. That's mostly for CI/CD enabled projects but I have seen teams use it manually as well.

**Reply by JoeyNovais** [reply]
Thanks for the answer and this awesome tool BTW ! :)

May I also suggest to add a note in the documentation around that point (the 3 ways of templating) ?

**Reply by JoernBerkefeld** [reply]
good point. added #392 for that

**Reply by JoernBerkefeld** [reply]
@JoeyNovais  - took long but this is fixed on develop branch now via PR #436
